test/test_unit.c

```diff
@@ -357,6 +370,50 @@ static int test_native_use_hint(void)
 #endif /* MLD_USE_NATIVE_POLY_USE_HINT_88 || MLD_USE_NATIVE_POLY_USE_HINT_32 \
         */
 
+#if defined(MLD_USE_NATIVE_POLY_CHKNORM)
+static int test_poly_chknorm_core(const int32_t *input, int32_t B,
+                                  const char *test_name)
+{
+  mld_poly input_poly;
+  uint32_t test_result, ref_result;
+
+  memcpy(input_poly.coeffs, input, MLDSA_N * sizeof(int32_t));
+
+  test_result = mld_poly_chknorm(&input_poly, B);
+  ref_result = mld_poly_chknorm_c(&input_poly, B);
+
+  if (test_result != ref_result)
+  {
+    fprintf(stderr, "FAIL: %s - result mismatch: native=%u, ref=%u\n",
+            test_name, test_result, ref_result);
+    return 1;
+  }
+
+  return 0;
+}
+
+static int test_native_poly_chknorm(void)
+{
+  int32_t test_data[MLDSA_N];
+  int i, B;
+
+  generate_i32_array_zeros(test_data, MLDSA_N);
+  CHECK(test_poly_chknorm_core(test_data, 0, "poly_chknorm_zeros_MIN_B") == 0);
+  CHECK(test_poly_chknorm_core(test_data, MLDSA_Q - REDUCE32_RANGE_MAX,
+                               "poly_chknorm_zeros_MAX_B") == 0);
+
+  for (i = 0; i < NUM_RANDOM_TESTS; i++)
+  {
+    generate_i32_array_ranged(test_data, MLDSA_N, -REDUCE32_RANGE_MAX,
+                              REDUCE32_RANGE_MAX);
+    generate_i32_ranged(&B, 0, MLDSA_Q - REDUCE32_RANGE_MAX + 1);
```

Comment:
I don't think we need to test this with a random bound. Instead test it with each bound that we use in practice: 
 - MLDSA_GAMMA1 - MLDSA_BETA
 - MLDSA_GAMMA2 - MLDSA_BETA
 - MLDSA_GAMMA2